ooni/tests/test_utils.py

```diff
@@ -77,3 +78,49 @@ def test_directory_usage(self):
         with open(os.path.join(tmp_dir, "subdir", "something.txt"), "w") as out_file:
             out_file.write("A"*1000)
         self.assertEqual(directory_usage(tmp_dir), 1000*2)
+
+class LoggingTests(unittest.TestCase):
+    def setUp(self):
+        self.dir = self.mktemp()
+        os.makedirs(self.dir)
+        self.name = "test.log"
+        self.path = os.path.join(self.dir, self.name)
+
+    @patch('ooni.settings.config')
+    def test_length_rotate(self, mock_config):
+        mock_config.basic.loglevel = 'DEBUG'
+        mock_config.advanced.debug = False
+        mock_config.basic.rotate = 'length'
+        mock_config.basic.rotate_length = '1M'
+
+        ooni_logger = log.OONILogger()
+        ooni_logger.start(logfile=self.path)
+
+        ooni_logger.debug("1"*(1024**2))
+
+        # This will trigger the rotation
+        ooni_logger.debug("2")
```

Comment:
Perhaps stupid question, but: you set `advanced.debug` to `False` above... so, why is the `debug()` function then sending logs to the logfile?